softgroup/services/profile/profile-impl/src/main/java/com/softgroup/profile/impl/handler/ContactsSyncHandler.java

```diff
@@ -17,12 +26,55 @@
 public class ContactsSyncHandler
         extends AbstractRequestHandler<ContactsSyncRequest,
                 ContactsSyncResponse> implements ProfileRequestHandler {
+
+    @Autowired
+    private ContactService contactService;
+
     public String getName() {
         return "contacts_sync";
     }
 
     @Override
     public Response<ContactsSyncResponse> doHandle(Request<ContactsSyncRequest> request) {
-        return null;
+        ContactsSyncRequest requestData = request.getData();
+        ContactsSyncResponse contactsSyncResponse = new ContactsSyncResponse();
+
+        String profileId = request.getRoutingData().getProfileId();
+        addContacts(requestData.getAddedContacts(), profileId);
+        removeContacts(requestData.getRemovedContacts(), profileId);
+
+        Response<ContactsSyncResponse> response = new Response<ContactsSyncResponse>();
+        response.setHeader(request.getHeader());
+        response.setData(contactsSyncResponse);
+
+        ResponseStatus status = new ResponseStatus();
+        status.setCode(200);
+        status.setMessage("OK");
+        response.setStatus(status);
+        return response;
+    }
+
+    private void addContacts(List<ContactDTO> addContacts, String profileId) {
+        for (ContactDTO addContact : addContacts) {
+            ContactEntity contactEntity =
+                    contactService.findByNameAndProfileId(addContact.getName(), profileId);
```

Comment:
@VadymPolyanski What if i have two contacts with same name but different phone numbers in my phone-book ?